src/app/core/services/scheduling.service.ts

```diff
@@ -220,6 +242,21 @@ export class SchedulingService {
       schedule = schedule.concat(tmpSchedule)
       scheduleLength = schedule.length
     }
+    // NOTE: Check for completed tasks
+    this.completedTasks.map(d => {
+      const index = schedule.findIndex(
+        s =>
+          ((this.utcOffsetPrev != null &&
+            s.timestamp - currentMidnight == d.timestamp - prevMidnight) ||
+            (this.utcOffsetPrev == null && s.timestamp == d.timestamp)) &&
+          s.name == d.name &&
+          !s.isClinical
+      )
+      if (index > -1) {
+        schedule[index].completed = true
+        return this.addToCompletedTasks(d)
```

Comment:
Should this be `s`? What's the point of mapping `d` to `d`?

Reply:
Since `SCHEDULE_TASKS_COMPLETED` is reset on schedule generation, this adds the tasks back (but only tasks from the midnight of when this function is run / new schedule is generated) in case of another timezone change or config reload in the future.

Reply:
But d is from an older schedule which will be discarded now. So do you think its better to add the corresponding task from the new generated schedule ?

Reply:
Yup it's fixed here: https://github.com/RADAR-base/RADAR-Questionnaire/pull/191/commits/cc3fcb214f7e9c0f582705dadcb8dd2f0e2d9213